lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -876,32 +892,53 @@ slotDifference epl a b
     a' = flatSlot epl a
     b' = flatSlot epl b
 
+-- | Return the slot immediately before the given slot.
+slotPred :: SlotParameters -> SlotId -> SlotId
```

Comment:
function is not used except tests I believe

Reply:
> function is not used except tests I believe

That's correct. However, these functions make it easier to express the following laws:
```hs
slotSucc . slotStartingAtOrJustBefore . utcTimePred . slotStartTime == id
slotPred . slotStartingAtOrJustAfter  . utcTimeSucc . slotStartTime == id
```